cloud: add release notes for 20230823 (#14573)
pingcap/docs: tidb-cloud/tidb-cloud-release-notes.md

This page lists the release notes of [TiDB Cloud](https://www.pingcap.com/tidb-cloud/) in 2023.

> ## August 23, 2023

Author: ```suggestion
## August 22, 2023
```

> ## August 22, 2023

Author: ```suggestion
## August 23, 2023
```

> ## August 22, 2023

Author: ```suggestion
## August 23, 2023
```

**General changes**

- Support Google Cloud [Private Service Connect](https://cloud.google.com/vpc/docs/private-service-connect) for [TiDB Dedicated](/tidb-cloud/select-cluster-tier.md#tidb-dedicated) clusters.

    You can now create a private endpoint and establish a secure connection to a TiDB Dedicated cluster hosted on Google Cloud.

    Key benefits:

>     - Intuitive operations: help you create a private endpoint with only several steps.
>     - Enhanced security: establish a secure connection to protect your data.
>     - Improved performance: provide low-latency and high-bandwidth connectivity.

Reviewer: ```suggestion
    - Intuitive operations: helps you create a private endpoint with only several steps.
    - Enhanced security: establishes a secure connection to protect your data.
    - Improved performance: provides low-latency and high-bandwidth connectivity.
```

>     For more information, see [Connect via Private Endpoint with Google Cloud](/tidb-cloud/set-up-private-endpoint-connections-on-google-cloud.md).

Author: ```suggestion
  For more information, see [Connect via Private Endpoint with Google Cloud](/tidb-cloud/set-up-private-endpoint-connections-on-google-cloud.md).
```
